### PR TITLE
setup OSSF Scorecard workflow

### PR DESCRIPTION
**Pull request by mmorel-35**
This also pin github-actions and add permissions per job or workflow.
Don’t worry about hash they are handled by renovate.

Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

File: .github/workflows/codeql-daily.yml [hunk 1]
```diff
@@ -8,35 +8,39 @@ on:
 
 jobs:
   analyze:
+    permissions:
+      actions: read  # for github/codeql-action/init to get workflow details
+      contents: read  # for actions/checkout to fetch code
```
Comment (trask) [inline .github/workflows/codeql-daily.yml:13]: does it make sense to move `contents: read` up to the top-level like in the others?

File: .github/workflows/backport.yml [hunk 2]
```diff
@@ -6,8 +6,13 @@ on:
         description: "The pull request # to backport"
         required: true
 
+permissions:  # added using https://github.com/step-security/secure-repo
```
Comment (trask) [inline .github/workflows/backport.yml:9]: I didn't follow how this is related to https://github.com/step-security/secure-repo?

File: .github/workflows/issue-management-stale-action.yml [hunk 3]
```diff
@@ -5,11 +5,17 @@ on:
     # hourly at minute 23
     - cron: "23 * * * *"
 
+permissions:  # added using https://github.com/step-security/secure-repo
+  contents: read
```
Comment (trask) [inline .github/workflows/issue-management-stale-action.yml:9]: should we add this to top-level of all workflows?

**Review by trask — COMMENTED** [review]
thx @mmorel-35, this looks great, left some basic questions I had below

cc @open-telemetry/sig-security-maintainers

File: .github/workflows/backport.yml (hunk 2 above)
Comment (mmorel-35) [reply .github/workflows/backport.yml:9]: I use https://app.stepsecurity.io/secureworkflow to help configure permissions and hash

File: .github/workflows/issue-management-stale-action.yml (hunk 3 above)
Comment (mmorel-35) [reply .github/workflows/issue-management-stale-action.yml]: That’s generally the OSSF recommandation indeed

**Review by codeboten — COMMENTED** [review]
Add this PR to the security SIG agenda for this week

**Comment by trask** [comment]
@mmorel-35 are we able to see the scorecard before merging this?

**Review by codeboten — APPROVED** [review]
This looks good, was discussed on 17-Jan-2024. Is there an output visible from running the ossf action somewhere?

**Comment by mmorel-35** [comment]
Yes, here is the version from my fork :
[OpenSSF Scorecard](https://securityscorecards.dev/viewer/?uri=github.com/mmorel-35/opentelemetry-java-instrumentation)

Once it’s good enough, you can add it to a readme.md with

```md
[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/open-telemetry/opentelemetry-java-instrumentation/badge)](https://securityscorecards.dev/viewer/?uri=github.com/open-telemetry/opentelemetry-java-instrumentation)
```

**Review by trask — APPROVED** [review]
thanks @mmorel-35!

**Comment by cartersocha** [comment]
Thanks for doing this @mmorel-35! Would you consider adding this to other repos? It’d be great to have wider coverage

**Comment by mmorel-35** [comment]
Hi @cartersocha ,
With pleasure, would like to suggest a list of projects you would have in mind ?

**Comment by cartersocha** [comment]
The main SDK, contrib, and Collector / Operator repos would be great. Most of this [list](https://github.com/orgs/open-telemetry/repositories?language=&q=&sort=stargazers&type=all) on the first page outside the OTEPs, community, docs, website, demo, and network repos. 

I don't want to over commit you. If it's relatively simple and you're time constrained we can implement this in the top couple repos, document the process, and then open issues in the other repos.

**Pull request merged**